src/main/scala/com/gu/scanamo/DerivedDynamoFormat.scala

```diff
@@ -55,3 +56,23 @@ trait DerivedDynamoFormat {
       def write(t: T): AttributeValue = formatR.value.write(gen.to(t))
     }
 }
+
+trait NoFormatError {
+  import scala.language.experimental.macros
+  implicit def moreHelpfulErrorMessage[T]: DynamoFormat[T] = macro MacroOfLastResort.moreHelpfulErrorMessage[T]
+}
+
+@bundle
```

Comment:
See http://docs.scala-lang.org/overviews/macros/bundles.html

Also get used by https://github.com/milessabin/macro-compat if you want cross-compilation.